pennylane_lightning/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -791,387 +640,6 @@ class StateVectorKokkos final
         }
     }
 
-    /**
-     * @brief Apply a PauliX operator to the state vector using a matrix
-     *
-     * @param wires Wires to apply gate to.
-     * @param inverse Indicates whether to use adjoint of gate.
-     * @param params Parameters for this gate
-     */
-    void applyPauliX(const std::vector<size_t> &wires, bool inverse = false,
```

Comment:
Aren't these exposed at the binding layer? With this change we always need a lookup to apply the gate, right?

Reply:
No, the gates are registered [here](https://github.com/PennyLaneAI/Lightning-Unification/blob/main/pennylane_lightning/src/bindings/BindingsBase.hpp#L58) via a loop creating lambdas calling `applyOperation`. In the Python [class](https://github.com/PennyLaneAI/Lightning-Unification/blob/main/pennylane_lightning/lightning_kokkos.py#L395), we are getting the lambda handle (if it exists) with the gate name.

Reply:
Oh, I wasn't aware that was happening. Ok, if this unifies the design we can follow through

Reply:
LQubit has multiple definitions of say `applyPauliX` but they exists to branch between kernels. It may not seem like it at first glance, but I think this indeed unifies the design. Now, like [LQubit](https://github.com/PennyLaneAI/Lightning-Unification/blob/main/pennylane_lightning/src/simulators/lightning_qubit/StateVectorLQubit.hpp#L72), LKokkos defines a map to gates, generators, etc.